### PR TITLE
mgr/telemetry: catch exception during requests.put

### PR DESCRIPTION
**Pull request by liewegas**
Sometimes requests.put throws an exception.  When that happens, generate
an appropriate error but don't crash the module.

Fixes: https://tracker.ceph.com/issues/43963
Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

**Comment by yaarith** [comment]
tested, looks good.
we should do the same for 'device' endpoint!

File: src/pybind/mgr/telemetry/module.py [hunk 1]
```diff
@@ -673,16 +673,23 @@ def send(self, report, endpoint=None):
...
-                        self.url,
-                        resp.status_code,
-                        resp.reason,
-                        resp.text
-                    ))
+                fail_reason = None
+                try:
+                    resp = requests.put(url=self.url, json=report,
+                                        proxies=proxies)
+                    if not resp.ok:
```
Comment (dmick) [inline src/pybind/mgr/telemetry/module.py]: should probably change to "raise_for_status()" and then handle everything as an exception; avoids duplicating code

**Comment by dmick** [comment]
LGTM

**Comment by yaarith** [comment]
looks good

**Pull request merged**